client/src/components/MobileMenu.tsx

```diff
@@ -58,26 +59,44 @@ export function MobileMenu({ handleSignInOrOut, handleNavClick }: Props) {
             <ul>
               {open &&
                 categories.map((category) => (
-                  <li key={category.id}>
-                    <Link to="/" onClick={() => handleNavClick(category.name)}>
+                  <li key={category.categoryId}>
+                    <Link
+                      to="/"
+                      onClick={() => fetchCategoryName(category.name)}>
                       {category.name}
                     </Link>
                   </li>
                 ))}
             </ul>
           </li>
-          <li className="rounded-lg mx-4 mb-4 mt-2 bg-accent-gray w-80">
-            {user && (
-              <Link to="/" onClick={handleSignInOrOut}>
-                Sign Out
-              </Link>
-            )}
-            {!user && (
+          {!user && (
+            <li className="rounded-lg mx-4 mb-4 mt-2 bg-accent-gray w-80">
               <Link to="/sign-in" onClick={handleSignInOrOut}>
                 Sign In
               </Link>
-            )}
-          </li>
+            </li>
+          )}
+          {user && (
+            <li className="rounded-lg mx-4 mb-4 mt-2 bg-accent-gray w-80">
+              <Link to="/create-post" onClick={handleMenuClick}>
+                Make a Post
+              </Link>
+            </li>
+          )}
+          {!user && (
+            <li className="rounded-lg mx-4 mb-4 mt-2 bg-accent-gray w-80">
+              <Link to="/sign-up" onClick={handleSignInOrOut}>
+                Register
+              </Link>
+            </li>
+          )}
+          {user && (
+            <li className="rounded-lg mx-4 mb-4 mt-2 bg-accent-gray w-80">
+              <Link to="/" onClick={handleSignInOrOut}>
+                Sign Out
+              </Link>
+            </li>
+          )}
```

Comment:
Why not combine these? You have two `li`'s that you want to render based on if there is a user or not a user, so rather than doing the check 4 separate times, why not do the check twice, and render both things you want?